whedon.rb

```diff
@@ -224,26 +232,44 @@ def assign_editor(new_editor)
   new_editor = new_editor.gsub(/^\@/, "")
   new_body = issue.body.gsub(/\*\*Editor:\*\*\s*(@\S*|Pending)/i, "**Editor:** @#{new_editor}")
   github_client.update_issue(@nwo, @issue_id, issue.title, new_body, :assignees => [])
-  update_assigness([new_editor])
+  reviewer_logins = reviewers.map { |reviewer_name| reviewer_name.sub(/^@/, "") }
+  update_assignees([new_editor] | reviewer_logins)
 end
 
 # Change the reviewer listed at the top of the issue
 def assign_reviewer(new_reviewer)
-  new_reviewer = new_reviewer.gsub(/^\@/, "")
-  editor = issue.body.match(/\*\*Editor:\*\*\s*.@(\S*)/)[1]
-  new_body = issue.body.gsub(/\*\*Reviewer:\*\*\s*(@\S*|Pending)/i, "**Reviewer:** @#{new_reviewer}")
-  github_client.add_collaborator(@nwo, new_reviewer)
-  puts "NWO: #{@nwo}"
-  puts "ISSUE ID: #{@issue_id}"
-  puts "TITLE: #{issue.title}"
-  puts "BODY: #{new_body}"
-  puts "ASSIGNEES #{[new_reviewer, editor]}"
+  set_reviewers([new_reviewer])
+end
+
+def add_reviewer(reviewer)
+  set_reviewers(reviewers + [reviewer])
+end
+
+def remove_reviewer(reviewer)
+  set_reviewers(reviewers - [reviewer])
+end
+
+def set_reviewers(reviewer_list)
+  reviewer_logins = reviewer_list.map { |reviewer_name| reviewer_name.sub(/^@/, "").downcase }.uniq
+  label = reviewer_list.empty? ? "Pending" : reviewer_list.join(", ")
+  new_body = issue.body.gsub(/\*\*Reviewers?:\*\*\s*(.+?)\r?\n/i, "**Reviewers:** #{label}\r\n")
+  reviewer_logins.each do |reviewer_name|
+    github_client.add_collaborator(@nwo, reviewer_name)
+  end
   github_client.update_issue(@nwo, @issue_id, issue.title, new_body, :assignees => [])
```

Comment:
We probably don't need to clear the list of assignees first now

Reply:
I think we still do. I took this for a spin in https://github.com/openjournals/joss-reviews-testing/issues/62 (with `whedon-api` deployed to a test Heroku instance) and when I tried removing the empty assignee param here things didn't work as hoped (i.e. the old review remains an assignee on the issue).

The reason for this (I believe) is that [this post to the GitHub API](https://github.com/openjournals/whedon-api/pull/24/files#diff-61dc6944d03f4f1695a11947c20e4a90R273) doesn't clobber existing assignees, it only adds new ones (see https://developer.github.com/v3/issues/assignees/#add-assignees-to-an-issue)